File: sdks/csharp/src/Sudomimus.Token/TokenException.cs
namespace Sudomimus.Token;

/// <summary>
/// Categorical reason a token failed to parse or verify.
/// </summary>
public enum TokenErrorCode
{
    InvalidJwt,
    WrongKeyType,
    MissingAudience,
    Expired,
    InvalidSignature,
}

/// <summary>
/// Thrown by <see cref="TokenParser"/> and <see cref="TokenVerifier"/> when
/// a token fails to parse or fails verification.
/// </summary>
public sealed class TokenException : Exception
{
    public TokenErrorCode Code { get; }

    public TokenException(TokenErrorCode code, string message)
        : base(message)
    {
        Code = code;
    }
}

File: sdks/java/token/src/main/java/com/sudomimus/token/TokenException.java
package com.sudomimus.token;

/** Thrown by {@link TokenParser} and {@link TokenVerifier} on parse or verification failure. */
public final class TokenException extends RuntimeException {

    private final TokenErrorCode code;

    public TokenException(TokenErrorCode code, String message) {
        super(message);
        this.code = code;
    }

    public TokenException(TokenErrorCode code, String message, Throwable cause) {
        super(message, cause);
        this.code = code;
    }

    public TokenErrorCode getCode() {
        return code;
    }
}
